sor/src/test/java/com/bazaarvoice/emodb/sor/core/CompactionControlTest.java

```diff
@@ -23,104 +24,178 @@
         Delta delta3 = Deltas.mapBuilder().put("key3", "change").build();
-        Compaction compaction1 = new Compaction(2, t4, t6, "abcdef0123456789", t6, t6, delta1);
 
-        final List<Map.Entry<UUID, Compaction>> compactions = ImmutableList.of(
-                Maps.immutableEntry(t4, compaction1));
+        final List<Map.Entry<UUID, Change>> deltas = ImmutableList.of(
+                Maps.immutableEntry(t1, ChangeBuilder.just(t1, delta1)),
+                Maps.immutableEntry(t2, ChangeBuilder.just(t2, delta2)),
+                Maps.immutableEntry(t3, ChangeBuilder.just(t3, delta3)));
+
+        Record record = mock(Record.class);
+        final Key key = mock(Key.class);
+        when(record.getKey()).thenReturn(key);
+        final List<Map.Entry<UUID, Compaction>> compactions = Lists.newArrayList();
+        when(record.passOneIterator()).thenReturn(compactions.iterator());
+        when(record.passTwoIterator()).thenReturn(deltas.iterator());
+
+        //noinspection unchecked
+        Supplier<Record> requeryFn = mock(Supplier.class);
+        when(requeryFn.get()).thenReturn(record);
+
+        MetricRegistry metricRegistry = new MetricRegistry();
+        Counter archiveDeltaSize = metricRegistry.counter(MetricRegistry.name("bv.emodb.sor", "DefaultCompactor", "archivedDeltaSize"));
+        Compactor compactor = new DistributedCompactor(archiveDeltaSize, false, metricRegistry);
+        Expanded expanded = compactor.expand(record, fullConsistencyTimestamp, fullConsistencyTimestamp, compactionControlTimestamp, MutableIntrinsics.create(key), false, requeryFn);
+
+        assertTrue(expanded.getPendingCompaction() != null);
+        // Do not delete deltas just yet; deltas are going to be deleted now in the next read
+        assertTrue(expanded.getPendingCompaction().getKeysToDelete().isEmpty());
+        // Add the compactions to the compaction list
+        UUID compactionTimestamp = TimeUUIDs.uuidForTimeMillis(nowInTimeMillis + 35 * 1000); // altering this instead of "expanded.getPendingCompaction().getChangeId()"
+        compactions.add(Maps.immutableEntry(compactionTimestamp, expanded.getPendingCompaction().getCompaction()));
+        when(record.passOneIterator()).thenReturn(compactions.iterator());
+        when(record.passTwoIterator()).thenReturn(deltas.iterator());
+
+        expanded = compactor.expand(record, fullConsistencyTimestamp, fullConsistencyTimestamp, compactionControlTimestamp, MutableIntrinsics.create(key), false, requeryFn);
+
+        // Verify that there are all 3 keys for deletion.
+        // In the DistributedCompactor, at first 3 keys will be selected for deletion, and later no keys get filtered out from the deletionList based on the compactionControlTimestamp.
+        assertTrue(expanded.getPendingCompaction() != null);
+        assertTrue(expanded.getPendingCompaction().getKeysToDelete().size() == 3, "All 3 deltas should up for deletion.");
```

Comment:
Minor:  Checking the size of `getKeysToDelete()` is redundant since the next line does an exact content comparison.